qiskit_dynamics/backend/dynamics_backend.py

```diff
@@ -351,6 +374,159 @@ def target(self) -> Target:
+            within these attributes will be extracted. If required, for backwards compatibility,
+            ``'configuration'`` and ``'defaults'`` options can be set, which will be returned via
+            the :meth:`.configuration` and :meth:`.defaults` methods.
+
+        The optional argument ``subsystem_list`` specifies which subset of qubits to model in the
+        constructed :class:`DynamicsBackend`. All other qubits are dropped from the model.
+
+        Configuration of the underlying :class:`.Solver` is controlled via the ``rotating_frame``,
+        ``evaluation_mode``, and ``rwa_cutoff_freq`` options. In contrast to :class:`.Solver`
+        initialization, ``rotating_frame`` defaults to the string ``"auto"``, which allows this
+        method to choose the rotating frame based on ``evaluation_mode``:
+
+        * If a dense evaluation mode is chosen, the rotating frame will be set to the
+          ``static_hamiltonian`` indicated by the Hamiltonian in ``backend.configuration()``.
+        * If a sparse evaluation mode is chosen, the rotating frame will be set to the diagonal of
+          ``static_hamiltonian``.
+
+        Otherwise the ``rotating_frame``, ``evaluation_mode``, and ``rwa_cutoff_freq`` are passed
+        directly to the :class:`.Solver` initialization.
+
+        Args:
+            backend: The ``Backend`` instance to build the :class:`.DynamicsBackend` from.
+            subsystem_list: The list of qubits in the backend to include in the model.
+            rotating_frame: Rotating frame argument for the internal :class:`.Solver`. Defaults to
+                ``"auto"``, allowing this method to pick a rotating frame.
+            evaluation_mode: Evaluation mode argument for the internal :class:`.Solver`.
+            rwa_cutoff_freq: Rotating wave approximation argument for the internal :class:`.Solver`.
+            **options: Additional options to be applied in construction of the
+                :class:`.DynamicsBackend`.
+
+        Returns:
+            DynamicsBackend
+
+        Raises:
+            QiskitError: If any required parameters are missing from the passed backend.
+        """
+
+        if not hasattr(backend, "configuration"):
+            raise QiskitError(
+                "DynamicsBackend.from_backend requires that the backend argument has a "
+                "configuration method."
+            )
+        if not hasattr(backend, "defaults"):
+            raise QiskitError(
+                "DynamicsBackend.from_backend requires that the backend argument has a defaults "
+                "method."
+            )
+
+        config = backend.configuration()
+        defaults = backend.defaults()
+
+        # get and parse Hamiltonian string dictionary
+        if subsystem_list is not None:
+            subsystem_list = sorted(subsystem_list)
+            if subsystem_list[-1] >= config.n_qubits:
```

Comment:
For V2 backend you should be able to do 
```
n_qubits = backend.target.num_qubits
...
            if subsystem_list[-1] >= n_qubits:
```

Reply:
You can also just do `backend.num_qubits` (https://qiskit.org/documentation/stubs/qiskit.providers.BackendV2.html#qiskit.providers.BackendV2.num_qubits)